Remove redundant `len` and `nil` check before loop

Juneezee opened the pull request:
## What does this PR change?

From the Go specification (https://go.dev/ref/spec#For_range):

> "1. For a nil slice, the number of iterations is 0."
> "3. If the map is nil, the number of iterations is 0."

`len` returns 0 if the slice or map is nil (https://pkg.go.dev/builtin#len). Therefore, having these checks before a loop is unnecessary:

- `len(v) != 0`
- `len(v) > 0`
- `v != nil`

## Does this PR relate to any other PRs?
No.

## How will this PR impact users?
No user-facing change.

## Does this PR address any GitHub or Zendesk issues?


## How was this PR tested?
`go build ./...` and `go test ./...`

## Does this PR require changes to documentation?
No.

## Have you labeled this PR and its corresponding Issue as "next release" if it should be part of the next OpenCost release? If not, why not?
No. External contributor cannot label PR.

Juneezee reviewed (commented):
Tip: Hide the whitespace to make the diff easier to read

![image](https://github.com/opencost/opencost/assets/20135478/418cbaa1-5f12-4b1a-ae8a-25d658af017d)

![image](https://github.com/opencost/opencost/assets/20135478/81a54c58-76b1-4948-a51f-0525f8778eb0)

AjayTripathy commented:
Thank you for this PR. I suspect these hooks were left in the code to quickly add an 

else {
  // log error
}

since many of these seem to imply exceptional cases except for this https://github.com/opencost/opencost/pull/2184/files#diff-084f229ce009968d50ab3e5cbeb8ff6f42eb0fd08fdc7b9d027f6931d47ea483R1765

Juneezee commented:
@cliffcolvin Friendly ping :smiley:

AjayTripathy commented:
Hey @Juneezee I think we'd rather log something about these values being nil in an else clause, as I mentioned earlier, than drop the if altogether since these cases feel exceptional.

Juneezee committed:
Remove redundant `len` and `nil` check before loop

`len` returns 0 if the slice or map is nil. From the Go specification
[1]:

  "1. For a nil slice, the number of iterations is 0."
  "3. If the map is nil, the number of iterations is 0."

Therefore, having these checks before a loop is unnecessary:

  - len(v) != 0
  - len(v) > 0
  - v != nil

[1]: https://go.dev/ref/spec#For_range

Signed-off-by: Eng Zer Jun <engzerjun@gmail.com>

Juneezee committed:
Add log for exceptional cases

Reference: https://github.com/opencost/opencost/pull/2184#issuecomment-1928551548
Signed-off-by: Eng Zer Jun <engzerjun@gmail.com>

Juneezee commented on pkg/cloud/aws/provider.go:
This one is redundant. I don't think we need a `log`.

Juneezee commented on pkg/costmodel/costmodel.go:
This one is redundant. I don't think we need a `log`.

Juneezee commented on pkg/costmodel/metrics.go:
This one is redundant. I don't think we need a `log`.

Juneezee reviewed (commented):
@AjayTripathy Updated. Please take a look again. Thanks!

mattray commented:
@AjayTripathy do you want to take a quick look?
